open_parser/base.py

```diff
@@ -8,32 +8,41 @@
 CAMBIO_EXTRACT_URL = (
     "https://p1iz3c1c77.execute-api.us-west-2.amazonaws.com/v1/cambio_api/extract"
 )
+CAMBIO_PARSE_URL = (
+    "https://p1iz3c1c77.execute-api.us-west-2.amazonaws.com/v1/cambio_api/parse"
+)
 
 
 class OpenParser:
-    def __init__(self, apiKey="") -> None:
+    def __init__(self, apiKey) -> None:
         self._uploadurl = CAMBIO_UPLOAD_URL
         self._extracturl = CAMBIO_EXTRACT_URL
+        self._parseurl = CAMBIO_PARSE_URL
         self._request_header = {"x-api-key": apiKey}
 
     def setAPIKey(self, apiKey):
         self._request_header = {"x-api-key": apiKey}
 
-    def extract_pdf_content(self, file_path):
+    def extract(self, file_path):
         user_id, job_id, s3_key = self._request_and_upload_by_apiKey(file_path)
         result = self._request_file_extraction(user_id, job_id, s3_key)
         return result["file_content"]
 
+    def parse(self, file_path, prompt):
+        user_id, job_id, s3_key = self._request_and_upload_by_apiKey(file_path, prompt)
+        result = self._request_info_extraction(user_id, job_id, s3_key)
+        return result["results"]
+
     def _error_handler(self, response):
         if response.status_code == 403:
             raise Exception("Invalid API Key")
         elif response.status_code == 429:
-            return Exception("API Key limit exceeded")
+            raise Exception("API Key limit exceeded")
         else:
-            return Exception(f"Error: {response.status_code} {response.text}")
+            raise Exception(f"Error: {response.status_code} {response.text}")
 
-    def _request_and_upload_by_apiKey(self, file_path):
-        params = {"fileName": file_path}
+    def _request_and_upload_by_apiKey(self, file_path, prompt=""):
```

Comment:
nit: we should force people to pass in prompt or to fail people early. Otherwise, user might force to put prompt and cause the backend to function improperly.